### DISCUSSION TITLE
Working Ubuntu USB Image

### DISCUSSION THREAD

**Discussion by bakerjalexander** [zbm-dev/zfsbootmenu#321] [General]
Hello all!

I have created a small (768M) bootable image using systemd-bootd files from Ubuntu Bionic and the latest ZFS Boot Menu files.

I have tested it with Ubuntu Bionic and ZFS version 2.1.5-1~18.04.york0 (from Jonathan F's repository) and it works fine. I don't see why it wouldn't work with just about any flavor of Ubuntu. There's enough space to add other kernels too, rather than just the latest.

To use it, simply dd it to a USB stick. I used "dd if=zfsbootmenu.img of=/dev/{your_drive} bs=64k conv=fdatasync

Now if I can just figure out how to send it to you. I guess it's too big to send here and I'm kind of a github novice.

Hope to hear from you! Thanks for your work on this. It's a great project !

**Comment by zdykstra** [comment]
Thank you for your interest in ZFSBootMenu! Instead of trying to copy a large image file, can you instead document the steps used to create it? What is the impetus behind creating this - what problem/problems does it solve that the normal EFI image doesn't solve?

**Comment by bakerjalexander** [comment]
To be honest, I created the image by copying an already made vfat EFI
partition and then erasing the existing kernel files and adding yours.

I did this because the instructions provided on the wiki about how to make
a bootable USB were not very clear and it seemed a lot easier just to do it
this way.

The nice thing about this is that because it has some additional space, you
can load up other kernels and boot those or you can simply boot into the
ZFS boot menu and use the default latest one.

Further, you can pass along boot line parameters to ZFSBootMenu by simply
adding them to the config file used to boot up the ZFSBootMenu kernel.

Finally, the USB stick should be able to be used by virtually any flavor of
Linux.

I was able to compress the image down to 430M which nowadays isn't so bad I
think. If you'd like me to send it to you, I'll be happy to do so.

Thanks again for all your help!

**Comment by ahesford** [comment]
I can see some value in systemd-boot or gummiboot as an alternative to rEFInd for loading the separate kernel/initramfs components that we distribute. Sometimes, having a bootloader that can adjust the command-line parameters passed to ZFSBootMenu or allow convenient selection of a fallback image is nice to have. A [wiki entry](https://github.com/zbm-dev/zfsbootmenu/wiki) describing the process for deploying systemd-boot alongside some ZFSBootMenu kernels on an external drive would be a welcome addition, if you care to write one.

As far as binary images, we have no convenient place to host them and, in any case, aren't willing to officially endorse any binary images that we don't create and sign ourselves.

**Comment by bakerjalexander** [comment]
Well I hadn't thought about the binary image problem but I do agree with
you.

Your project was a big help to me. It actually helped me isolate a boot
hang problem I had with a newer version of ZFS and I think it will be very
useful in trying to upgrade my servers from old versions of Ubuntu to newer
ones.

In short, I'll be glad to write that systemd-bootd tutorial up for you as
my way of saying thanks for your development efforts. I'll work on it this
weekend.

Thanks!

**Comment by bakerjalexander** [comment]
Submitted for your review. Written in dash for portability. I have tested
this on my own system (Ubuntu Bionic) and it is working.

Let me know what you think.

Thanks!

**Comment by bakerjalexander** [comment]
Just wanted to check with you did you find the script useful?

**Comment by bakerjalexander** [comment]
Any update on this? I can just post to a new thread if you want

**Comment by ahesford** [comment]
Am I missing something? I don't see any script in this thread.

**Comment by bakerjalexander** [comment]
Probably my fault I think I've been conversing with two different people.
Anyway, I started a new discussion and attached the script there.

My Apologies,

John Baker
